lib/ohai/dsl/plugin.rb

```diff
@@ -183,6 +185,15 @@ def method_missing(name, *args)
 
       private
 
+      def safe_get_attribute(*keys)
+        attrs = @data
+        keys.each do |key|
+          return nil unless attrs.respond_to?(:has_key?) && attrs.has_key?(key)
```

Comment:
Ahh that is so nice. It breaks in the case where some intermediate key is not a hash, though. Hm...